script.js

```diff
@@ -0,0 +1,30 @@
+document.getElementById('check-btn').addEventListener('click', () => {
+  const userInput = document.getElementById('user-input').value.trim();
+  const resultsDiv = document.getElementById('results-div');
+
+  if (userInput === '') {
+    resultsDiv.textContent = 'Please provide a phone number';
+    return;
+  }
+
+  const validPatterns = [
+    /^1?\s?\d{3}[-\s]?\d{3}[-\s]?\d{4}$/,
+    /^1?\s?\(\d{3}\)\s?\d{3}[-\s]?\d{4}$/,
+    /^1?\s?\(\d{3}\)\d{3}[-\s]?\d{4}$/,
+    /^1?\(\d{3}\)\d{3}[-\s]?\d{4}$/,
+    /^\d{10}$/,
+  ];
+
+  const isValid = validPatterns.some((pattern) => pattern.test(userInput));
+
+  if (isValid) {
+    resultsDiv.textContent = `Valid US number: ${userInput}`;
+  } else {
+    resultsDiv.textContent = `Invalid US number: ${userInput}`;
+  }
+});
+
+document.getElementById('clear-btn').addEventListener('click', () => {
+  document.getElementById('user-input').value = '';
+  document.getElementById('results-div').textContent = '';
+});
```

Comment:
- [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  // As an Example I added only one Event Listener example here, you add your own:
  
  const clearBtn = document.getElementById('clear-btn');
  const resultsDiv = document.getElementById('results-div');
  
  clearBtn.addEventListener('click', () => {
    resultsDiv.innerHTML = '';
  });
})
```